scripts/investigate-houjin-quality.py

```
@@ -0,0 +1,460 @@
+
+if not os.path.exists(ZIP_PATH):
+    print('  ⚠ NTA ZIP が見つかりません（スキップ）')
+    nta3_by_cn: dict[str, str] = {}
+else:
+    print(f'  ZIP 読み込み中: {ZIP_PATH}')
+    # RS CSV に出現する有効法人番号セット
+    rs_cn_set = set(cn for cn in cn_info if re.fullmatch(r'\d{13}', cn))
+    nta3_by_cn: dict[str, str] = {}
+
+    with zipfile.ZipFile(ZIP_PATH, 'r') as zf:
+        csv_name = [n for n in zf.namelist() if n.endswith('.csv')][0]
```

Ruff (0.15.2)

[warning] Prefer `next(n for n in zf.namelist() if n.endswith('.csv'))` over single element slice

Replace with `next(n for n in zf.namelist() if n.endswith('.csv'))`

(RUF015)